phoenix-core/src/it/java/org/apache/phoenix/end2end/OnDuplicateKey2IT.java

```diff
@@ -198,37 +199,85 @@ public void testReturnRowResult() throws Exception {
+                            + "'pk005', 23, 'col2_001')";
+            conn.createStatement().execute(upsertSql);
+            upsertSql =
+                    "UPSERT INTO " + tableName
+                            + " (PK1, PK2, PK3, COUNTER1, COUNTER2) VALUES('pk003', 122.34, "
+                            + "'pk005', 23, 'col2_001')";
+            conn.createStatement().execute(upsertSql);
+
+            ps = conn.prepareStatement(
+                    "DELETE FROM " + tableName + " WHERE PK1 = ? AND PK2 = ?")
+                    .unwrap(PhoenixPreparedStatement.class);
+            ps.setString(1, "pk001");
+            ps.setDouble(2, 122.34);
+            validateReturnedRowAfterDelete(conn, ps, tableName, 2233.99, "col2_001", false,
+                    bsonDocument2, 234);
+
+            ps = conn.prepareStatement(
+                            "DELETE FROM " + tableName).unwrap(PhoenixPreparedStatement.class);
+            validateReturnedRowAfterDelete(conn, ps, tableName, 2233.99, "col2_001", false,
+                    bsonDocument2, 234);
+
+            ResultSet rs = conn.createStatement().executeQuery("SELECT * FROM " + tableName);
+            assertFalse(rs.next());
         }
     }
 
-    private static void validateReturnedRowAfterUpsert(Connection conn,
-                                                       String upsertSql,
+    private static void validateReturnedRowAfterDelete(Connection conn,
+                                                       PhoenixPreparedStatement ps,
                                                        String tableName,
                                                        Double col1,
                                                        String col2,
                                                        boolean success,
-                                                       BsonDocument inputDoc,
                                                        BsonDocument expectedDoc,
                                                        Integer col4)
             throws SQLException {
-        final Pair<Integer, Tuple> resultPair;
-        if (inputDoc != null) {
-            PhoenixPreparedStatement ps =
-                    conn.prepareStatement(upsertSql).unwrap(PhoenixPreparedStatement.class);
-            ps.setObject(1, inputDoc);
-            resultPair = ps.executeUpdateReturnRow();
-        } else {
-            resultPair = conn.createStatement().unwrap(PhoenixStatement.class)
-                    .executeUpdateReturnRow(upsertSql);
-        }
-        assertEquals(success ? 1 : 0, resultPair.getFirst().intValue());
+        final Pair<Integer, Tuple> resultPair = ps.executeUpdateReturnRow();
         ResultTuple result = (ResultTuple) resultPair.getSecond();
+
         PTable table = conn.unwrap(PhoenixConnection.class).getTable(tableName);
 
+        if (!success && result.getResult() == null) {
+            return;
```

Comment:
I couldn't understand when will the result be null ?

Reply:
It will be null when Delete does not have strict point lookup.